### PR TITLE
Adding support for doing regression tests

### PR DESCRIPTION
**Pull request by ndtrung81**
**Summary**

This PR adds a Python script for performing regression tests of a chosen LAMMPS binary using the log files under the examples/ subfolders as reference. Details and example uses are given in the README file in `tools/regression-tests`.

With the current features, users can:
    + specify which LAMMPS binary version to test (e.g., the version from a commit, or those from `lammps-testing`)
    + specify the examples subfolders (thus the reference log files) seperately (e.g. from other LAMMPS versions or commits)
    + specify tolerances for individual quantities for any input script to override the global values
    + launch tests with `mpirun` with all supported command line features (multiple procs, multiple paritions, and suffixes, or with `valgrind`)
    + skip certain input files (whose names match specified patterns) if not interested, or packaged not installed, or no reference log file exists
    + simplify the main LAMMPS builds, as long as a LAMMPS binary is available
    + keep track of the testing progress to resume the testing from the last checkpoint (skipping completed runs)
    + split the list of input scripts under the top level examples into multiple lists and launch separate runs (there are 800+ input script under the top-level examples)

Limitations:
    - input scripts use thermo style multi (e.g., examples/peptide) do not work with the expected thermo output format
    - input scripts that require partition runs (e.g. examples/neb) need a separate config file, e.g. args: `--partition 3x1`
    - testing accelerator packages (GPU, INTEL, KOKKOS, OPENMP) need separate config files, `args: -sf omp -pk omp 4`

TODO:
    + be able to be invoked from run_tests in the lammps-testing infrastruture

The following Python packages need to be installed into an activated environment:
    
    python3 -m venv testing-env
    source testing-env/bin/activate
    pip install numpy pyyaml junit_xml

Example usage:

1) Simple use (using the provided tools/regression-tests/config.yaml and the examples/ folder at the top level)

    python3 run_tests.py --lmp-bin=/path/to/lmp_binary

2) Use a custom testing configuration
           
    python3 run_tests.py --lmp-bin=/path/to/lmp_binary --config-file=/path/to/config/file/config.yaml

3) Specify a list of example folders
           
    python3 run_tests.py --lmp-bin=/path/to/lmp_binary --config-file=/path/to/config/file/config.yaml \
                --example-folders="/path/to/examples/folder1;/path/to/examples/folder2"

   The example folders can also be loaded from a text file list_subfolders1.txt:
           
    python3 run_tests.py --lmp-bin=/path/to/lmp_binary --config-file=/path/to/config/file/config.yaml \
                --list-input=list_subfolders1.txt --output-file=output1.txt --progress-file=progress1.yaml \
                --log-file=run1.log
          
4) Test a LAMMPS binary with the whole top-level /examples folder in a LAMMPS source tree
    
     python3 run_tests.py --lmp-bin=/path/to/lmp_binary --examples-top-level=/path/to/lammps/examples

5) Analyze the LAMMPS binary annd whole top-level /examples folder in a LAMMPS source tree and generate separate input lists for 8 workers:
           
     python3 run_tests.py --lmp-bin=/path/to/lmp_binary --examples-top-level=/path/to/lammps/examples \
                --analyze --num-workers=8

This is used for splitting the subfolders into separate input lists and launching different instances of run_tests.py simultaneously.

**Related Issue(s)**

N/A

**Author(s)**

Trung Nguyen (U Chicago)

**Licensing**

By submitting this pull request, I agree, that my contribution will be included in LAMMPS and redistributed under either the GNU General Public License version 2 (GPL v2) or the GNU Lesser General Public License version 2.1 (LGPL v2.1).

**Backward Compatibility**

No issue

**Implementation Notes**

* Need nicer-looking summary/output
* Support more output formats if necessary
* Code size grows really fast to handle exceptions/error'ing out runs, in particular the key function `iterate()` over the list of input scripts

**Post Submission Checklist**

- [ ] The feature or features in this pull request is complete
- [x] Licensing information is complete
- [x] Corresponding author information is complete
- [ ] The source code follows the LAMMPS formatting guidelines
- [ ] Suitable new documentation files and/or updates to the existing docs are included
- [ ] The added/updated documentation is integrated and tested with the documentation build system
- [ ] The feature has been verified to work with the conventional build system
- [ ] The feature has been verified to work with the CMake based build system
- [ ] Suitable tests have been added to the unittest tree.
- [x] A package specific README file has been included or updated
- [] One or more example input decks are included

**Further Information, Files, and Links**

### CODE REVIEW

**Comment by sjplimp** [comment]
@ndtrung81 why is there a brokenDump file added to examples/bpm/impact ?  Is that intentional?

File: tools/regression-tests/Cu_u3.eam
Comment (rbberger) [inline tools/regression-tests/Cu_u3.eam:1]: This file shouldn't be added again.

File: tools/regression-tests/data.rhodo
Comment (rbberger) [inline tools/regression-tests/data.rhodo:1]: Same for this file.

Comment (ndtrung81) [reply tools/regression-tests/data.rhodo:1]: Thanks for pointing out those duplicated files. I have removed them. The files were there initially for debugging.

**Comment by ndtrung81** [comment]
@sjplimp the changes to the file `example/bpm/impact/brokenDump` were unintentional in one of my previous commits ([9c7b169](https://github.com/lammps/lammps/pull/4228/commits/9c7b169003f7ac55772b0d6ace6d022c5c1c1b57)).Thanks for pointing this out. I reverted those changes in this file. (The file seems to be for debugging and should be removed anyway).

**Comment by sjplimp** [comment]
@jtclemm See the comment above about the examples/bpm/impact/brokenDump file being in the repo.  Normally we
don't put dump files in the repo.  But if it is small and you think it is useful for the user, then you could include it.  But it should have a name with a date stamp so that users can compare their run to the archived "correct" result, similar to log files.

**Comment by jtclemm** [comment]
> @jtclemm See the comment above about the examples/bpm/impact/brokenDump file being in the repo. Normally we don't put dump files in the repo. But if it is small and you think it is useful for the user, then you could include it. But it should have a name with a date stamp so that users can compare their run to the archived "correct" result, similar to log files.

I don't think there's any need to keep that file. Its inclusion was almost certainly just oversight.

**Comment by ndtrung81** [comment]
Version:                                 develop / patch_27Jun2024-446-g18b3f6b4d4 (SHA1 ID: 18b3f6b4d441837493061d7ef6ad3f629a99a1f8)
Compiler:                               mpich/4.1.2+gcc/10.2.0, built with sanitizer
CMake build: `-C ../cmake/presets/most.cmake`
Total input scripts examined:  808                                                                  
Completed runs:                     488 
Non-completed runs due to: 1) packages not installed, 2) wrong path to potential files, 3) lost atoms, 4) a preliminary run needed, 5) coupling with other codes (mdi, python) or interactive (imd, ipi),  6) multiple partitions needed

Details: (to be updated upon manual inspection of the individual output)


ASPHERE: in.line and in.line.srd `Lost atoms`
grid/in.grid.2d and in.sph
```
Caught signal 11 (Segmentation fault: address not mapped to object at address 0x30)
 ==== backtrace (tid:2890274) ====
  0 0x0000000000012b20 .annobin_sigaction.c()  sigaction.c:0
  1 0x0000000003200508 LAMMPS_NS::Grid2d::box_drop_grid()  /project/rcc/trung/lammps-git/src/grid2d.cpp:1815
  2 0x000000000320103c LAMMPS_NS::Grid2d::box_drop()  /project/rcc/trung/lammps-git/src/grid2d.cpp:1769
  3 0x00000000032069df LAMMPS_NS::Grid2d::compute_overlap()  /project/rcc/trung/lammps-git/src/grid2d.cpp:1678
  4 0x00000000032079cb LAMMPS_NS::Grid2d::setup_remap()  /project/rcc/trung/lammps-git/src/grid2d.cpp:1367
  5 0x0000000002e14d28 LAMMPS_NS::FixAveGrid::reset_grid()  /project/rcc/trung/lammps-git/src/fix_ave_grid.cpp:2064
  6 0x00000000035049ca LAMMPS_NS::Modify::reset_grid()  /project/rcc/trung/lammps-git/src/modify.cpp:806
  7 0x0000000002e7de78 LAMMPS_NS::FixBalance::rebalance()  /project/rcc/trung/lammps-git/src/fix_balance.cpp:336
  8 0x0000000002e7f409 LAMMPS_NS::FixBalance::setup_pre_exchange()  /project/rcc/trung/lammps-git/src/fix_balance.cpp:215
  9 0x0000000003503ab9 LAMMPS_NS::Modify::setup_pre_exchange()  /project/rcc/trung/lammps-git/src/modify.cpp:330
 10 0x0000000003af5e01 LAMMPS_NS::Verlet::setup()  /project/rcc/trung/lammps-git/src/verlet.cpp:116
 11 0x0000000003963c36 LAMMPS_NS::Run::command()  /project/rcc/trung/lammps-git/src/run.cpp:171
 12 0x000000000331b360 LAMMPS_NS::Input::execute_command()  /project/rcc/trung/lammps-git/src/input.cpp:868
 13 0x0000000003320ad7 LAMMPS_NS::Input::file()  /project/rcc/trung/lammps-git/src/input.cpp:313
 14 0x00000000004023de main()  /project/rcc/trung/lammps-git/src/main.cpp:78
 15 0x0000000000023493 __libc_start_main()  ???:0
 16 0x000000000040256e _start()  ???:0
 =================================
 AddressSanitizer:DEADLYSIGNAL
 =================================================================
 ==2890274==ERROR: AddressSanitizer: SEGV on unknown address 0x15dbb19e002c1a22 (pc 0x7fa5826ae508 bp 0x7ffd1a225390 sp 0x7ffd1a224f70 T0)
```
grid/in.grid.3d:
```
Caught signal 11 (Segmentation fault: address not mapped to object at address 0x28)
 ==== backtrace (tid:2890228) ====
  0 0x0000000000012b20 .annobin_sigaction.c()  sigaction.c:0
  1 0x000000000321d398 LAMMPS_NS::Grid3d::box_drop_grid()  /project/rcc/trung/lammps-git/src/grid3d.cpp:2011
  2 0x000000000321df9c LAMMPS_NS::Grid3d::box_drop()  /project/rcc/trung/lammps-git/src/grid3d.cpp:1965
  3 0x0000000003225f6f LAMMPS_NS::Grid3d::compute_overlap()  /project/rcc/trung/lammps-git/src/grid3d.cpp:1860
  4 0x000000000322759d LAMMPS_NS::Grid3d::setup_remap()  /project/rcc/trung/lammps-git/src/grid3d.cpp:1540
 ==== backtrace (tid:2890231) ====
  0 0x0000000000012b20 .annobin_sigaction.c()  sigaction.c:0
  1 0x000000000321d398 LAMMPS_NS::Grid3d::box_drop_grid()  /project/rcc/trung/lammps-git/src/grid3d.cpp:2011
  2 0x000000000321df9c LAMMPS_NS::Grid3d::box_drop()  /project/rcc/trung/lammps-git/src/grid3d.cpp:1965
  3 0x0000000003225f6f LAMMPS_NS::Grid3d::compute_overlap()  /project/rcc/trung/lammps-git/src/grid3d.cpp:1860
  4 0x000000000322759d LAMMPS_NS::Grid3d::setup_remap()  /project/rcc/trung/lammps-git/src/grid3d.cpp:1540
  5 0x0000000002e1422c LAMMPS_NS::FixAveGrid::reset_grid()  /project/rcc/trung/lammps-git/src/fix_ave_grid.cpp:2066
  6 0x00000000035049ca LAMMPS_NS::Modify::reset_grid()  /project/rcc/trung/lammps-git/src/modify.cpp:806
  7 0x0000000002e7de78 LAMMPS_NS::FixBalance::rebalance()  /project/rcc/trung/lammps-git/src/fix_balance.cpp:336
  8 0x0000000002e7f409 LAMMPS_NS::FixBalance::setup_pre_exchange()  /project/rcc/trung/lammps-git/src/fix_balance.cpp:215
  9 0x0000000003503ab9 LAMMPS_NS::Modify::setup_pre_exchange()  /project/rcc/trung/lammps-git/src/modify.cpp:330
 10 0x0000000003af5e01 LAMMPS_NS::Verlet::setup()  /project/rcc/trung/lammps-git/src/verlet.cpp:116
 11 0x0000000003963c36 LAMMPS_NS::Run::command()  /project/rcc/trung/lammps-git/src/run.cpp:171
 12 0x000000000331b360 LAMMPS_NS::Input::execute_command()  /project/rcc/trung/lammps-git/src/input.cpp:868
 13 0x0000000003320ad7 LAMMPS_NS::Input::file()  /project/rcc/trung/lammps-git/src/input.cpp:313
 14 0x00000000004023de main()  /project/rcc/trung/lammps-git/src/main.cpp:78
 15 0x0000000000023493 __libc_start_main()  ???:0
 16 0x000000000040256e _start()  ???:0
 =================================
  5 0x0000000002e1422c LAMMPS_NS::FixAveGrid::reset_grid()  /project/rcc/trung/lammps-git/src/fix_ave_grid.cpp:2066
  6 0x00000000035049ca LAMMPS_NS::Modify::reset_grid()  /project/rcc/trung/lammps-git/src/modify.cpp:806
  7 0x0000000002e7de78 LAMMPS_NS::FixBalance::rebalance()  /project/rcc/trung/lammps-git/src/fix_balance.cpp:336
  8 0x0000000002e7f409 LAMMPS_NS::FixBalance::setup_pre_exchange()  /project/rcc/trung/lammps-git/src/fix_balance.cpp:215
  9 0x0000000003503ab9 LAMMPS_NS::Modify::setup_pre_exchange()  /project/rcc/trung/lammps-git/src/modify.cpp:330
 10 0x0000000003af5e01 LAMMPS_NS::Verlet::setup()  /project/rcc/trung/lammps-git/src/verlet.cpp:116
 11 0x0000000003963c36 LAMMPS_NS::Run::command()  /project/rcc/trung/lammps-git/src/run.cpp:171
 12 0x000000000331b360 LAMMPS_NS::Input::execute_command()  /project/rcc/trung/lammps-git/src/input.cpp:868
 13 0x0000000003320ad7 LAMMPS_NS::Input::file()  /project/rcc/trung/lammps-git/src/input.cpp:313
 14 0x00000000004023de main()  /project/rcc/trung/lammps-git/src/main.cpp:78
 15 0x0000000000023493 __libc_start_main()  ???:0
 16 0x000000000040256e _start()  ???:0
 =================================
 AddressSanitizer:DEADLYSIGNAL
 AddressSanitizer:DEADLYSIGNAL
 =================================================================
 =================================================================
 ==2890228==ERROR: AddressSanitizer: SEGV on unknown address 0x15dbb19e002c19f4 (pc 0x7fd40b53f398 bp 0x7fff10c88fb0 sp 0x7fff10c88a90 T0)
 ==2890231==ERROR: AddressSanitizer: SEGV on unknown address 0x15dbb19e002c19f7 (pc 0x7fe038a6d398 bp 0x7ffca4f463d0 sp 0x7ffca4f45eb0 T0)
 ==2890228==The signal is caused by a READ memory access.
 ==2890231==The signal is caused by a READ memory access.
```
COUPLE: skipped
HEAT/in.lj.exhex: need 32 nprocs (need to reduce np)
fire/: need partition runs -- manual, passed
imd/: interative runs with VMD (skipped)
kim/: need -DPKG_KIM=on (skipped)
mdi/: need -DPKG_MDI=on (skipped)
mliap/in.mliap.ace.compute 
```
ERROR: Illegal compute mliap command (src/ML-IAP/compute_mliap.cpp:110)
Last command: compute     ace all mliap descriptor ace H_N_O_ccs.yace model linear gradgradflag 1
```
mliap/in.mliap.pytorch.*: ML-IAP needs python support (skipped) 
in.mliap.quadratic.compute: failed, AddressSanitizer catches error
```
==2889350==ERROR: AddressSanitizer: heap-buffer-overflow on address 0x6120000137a8 at pc 0x7f224156d5c7 bp 0x7ffe90e9f570 sp 0x7ffe90e9f568
 WRITE of size 4 at 0x612000013c28 thread T0
     #0 0x7fe5fa0c35c6 in LAMMPS_NS::MLIAPData::generate_neighdata(LAMMPS_NS::NeighList*, int, int) /project/rcc/trung/lammps-git/src/ML-IAP/mliap_data.cpp:188
     #1 0x7fe5fa0bcf65 in LAMMPS_NS::ComputeMLIAP::compute_array() /project/rcc/trung/lammps-git/src/ML-IAP/compute_mliap.cpp:229
     #2 0x7fe5f7d1bc5d in LAMMPS_NS::FixAveTime::invoke_vector(long) /project/rcc/trung/lammps-git/src/fix_ave_time.cpp:741
     #3 0x7fe5f7d21d0d in LAMMPS_NS::FixAveTime::end_of_step() /project/rcc/trung/lammps-git/src/fix_ave_time.cpp:506
     #4 0x7f223f1bd2ec in LAMMPS_NS::FixAveTime::setup(int) /project/rcc/trung/lammps-git/src/fix_ave_time.cpp:492
     #5 0x7fe5f83c266d in LAMMPS_NS::Modify::setup(int) /project/rcc/trung/lammps-git/src/modify.cpp:317
     #6 0x7f223fe62ef1 in LAMMPS_NS::Verlet::setup(int) /project/rcc/trung/lammps-git/src/verlet.cpp:159
     #7 0x7fe5f8825c35 in LAMMPS_NS::Run::command(int, char**) /project/rcc/trung/lammps-git/src/run.cpp:171
     #8 0x7fe5f81dd35f in LAMMPS_NS::Input::execute_command() /project/rcc/trung/lammps-git/src/input.cpp:868
     #9 0x7fe5f81e2ad6 in LAMMPS_NS::Input::file() /project/rcc/trung/lammps-git/src/input.cpp:313
     #10 0x4023dd in main /project/rcc/trung/lammps-git/src/main.cpp:78
     #11 0x7f2238882492 in __libc_start_main (/lib64/libc.so.6+0x23492)
     #12 0x40256d in _start (/project/rcc/trung/lammps-git/build/lmp+0x40256d)
```
multi/in.powerlaw: `ERROR: Read_data zlo/zhi for 2d simulation must straddle 0.0 (src/read_data.cpp:538) ` (fixed by https://github.com/lammps/lammps/commit/462e9756ce066b3a6a95175c6bf837e3d43c4ef8)
neb/: need partition runs -- manual, 8 procs with -partition 4x2 failed with 3 example inputs
plumed: need -DPKG_PLUMED (skipped)
python/: need -DPKG_PYTHON=on
QUANTUM: need -DPKG_MDI=on (skipped)
SPIN/read_restart/in.spin.read_data -nan and failed with 1 proc
SPIN/read_restart/in.spin.read_data -nan and failed with 1 proc
SPIN/test_problems/validation_nvt/in.spin.nvt_spin, `ERROR: Fix langevin period must be > 0.0 (src/fix_langevin.cpp:80)`
SPIN/gneb/in.gneb.ironprd: need partition runs -- failed (4x1), nan (3x1)
SPIN/gneb/in.gneb.skyrmionprd: need partition runs -- failed
SPIN/setforce_spin/in.spin.setforce: failed (-np 1 and 4)
tad/: need partition runs -- manual, passed
template/in.mol-restart-mix: no molecular-mix.restart, need to write out from the run with in.molecular-mix (fixed by 8a77949db22c5a96d81b9dd55ec620b6df77434d)
template/in.tmpl-restart-mix: template-mix.restart, need to write out from the run with in.template-mix (fixed by 8a77949db22c5a96d81b9dd55ec620b6df77434d)
voronoi/: non standard log (no Step) --manual runs, passed

PACKAGES/adios: needs -DPKG_ADIOS=on (skipped)
PACKAGES/alchemy: need partition runs, need a README -- manual, passed
PACKAGES/atc: -DPKG_ATC=on (skipped)
PACKAGES/basal/in.basal too long for an example, no reference log file
PACKAGES/diffraction/in.BulkNi: `ERROR: Must set 'extvector' or 'extlist' when setting 'vector_flag' for fix saed/vtk.  Contact the developer. (src/fix.cpp:138)`
PACKAGES/dpd-smooth/2d-diffusion/in.lammps: `ERROR: Create_box region zlo/zhi for 2d simulation must straddle 0.0 (src/create_box.cpp:100) `(fixed by 2115ac8971ad4d4a22bbd995adec2174f3697059)
PACKAGES/dpd-smooth/2d-diffusion-in-shear-flow/in.lammps:  `ERROR: Create_box region zlo/zhi for 2d simulation must straddle 0.0 (src/create_box.cpp:100)` (fixed by 2115ac8971ad4d4a22bbd995adec2174f3697059)
PACKAGES/electrode/: in.conp, in.conq, in.conq2, in.etypes, in.ffield, in.ramp, in.thermo AddressSanitizer reported errors
  ```
AddressSanitizer: global-buffer-overflow on address 0x7fe1788c4c40 at pc 0x7fe175b50307 bp 0x7ffe0475a8e0 sp 0x7ffe0475a8d8
  READ of size 8 at 0x7fbfb0f3ec40 thread T0
      #0 0x7fbfae1ca306 in s_lmp_cat /project/rcc/trung/lammps-git/lib/linalg/s_lmp_cat.cpp:11
      #1 0x7fbfae1c7af4 in dtrtri_ /project/rcc/trung/lammps-git/lib/linalg/dtrtri.cpp:65
      #2 0x7fbfae1bf7ed in dgetri_ /project/rcc/trung/lammps-git/lib/linalg/dgetri.cpp:59
      #3 0x7fbfab85670a in LAMMPS_NS::FixElectrodeConp::invert() /project/rcc/trung/lammps-git/src/ELECTRODE/fix_electrode_conp.cpp:673
      #4 0x7fbfab881b2d in LAMMPS_NS::FixElectrodeConp::setup_post_neighbor() /project/rcc/trung/lammps-git/src/ELECTRODE/fix_electrode_conp.cpp:597
      #5 0x7fbfaaa6169c in LAMMPS_NS::Modify::setup_post_neighbor() /project/rcc/trung/lammps-git/src/modify.cpp:358
      #6 0x7fbfab054435 in LAMMPS_NS::Verlet::setup(int) /project/rcc/trung/lammps-git/src/verlet.cpp:130
      #7 0x7fbfaaec1c35 in LAMMPS_NS::Run::command(int, char**) /project/rcc/trung/lammps-git/src/run.cpp:171
      #8 0x7fbfaa87935f in LAMMPS_NS::Input::execute_command() /project/rcc/trung/lammps-git/src/input.cpp:868
      #9 0x7fbfaa87ead6 in LAMMPS_NS::Input::file() /project/rcc/trung/lammps-git/src/input.cpp:313
      #10 0x4023dd in main /project/rcc/trung/lammps-git/src/main.cpp:78
      #11 0x7fbfa3a74492 in __libc_start_main (/lib64/libc.so.6+0x23492)
      #12 0x40256d in _start (/project/rcc/trung/lammps-git/build/lmp+0x40256d)
```
PACKAGES/eff/CH4/in.ch4_ionized.dynamics:
`ERROR: Illegal pair style command (src/EFF/pair_eff_cut.cpp:828)`
PACKAGES/eff/ECP/Si2H6/in.Si2H6.ang:
`ERROR: Need to specify ECP type on pair_style command (src/EFF/pair_eff_cut.cpp:889)`
Last command: minimize        0 1.0e-6 2000 10000
PACKAGES/eff/Auger-Adamantane/in.adamantane_ionized.nve: `Lost atoms`

PACKAGES/electron_stopping/in.cascade_AlCu: `ERROR on proc 0: cannot open eam/alloy potential file ../../../../potentials/AlCu.eam.alloy: No such file or directory (src/potential_file_reader.cpp:58)` (need to create symlinks to those in ../../potentials, like in manybody examples)
PACKAGES/fep/ta/in.spce.lmp takes long (mpirun -np 12)
PACKAGES/fep/CH4-CF4/fep01 and fep10, bar10, bar01: need log files with right format with 12 procs, same for other subfoldders
PACKAGES/grem/lj-6rep/in.gREM: Need to run the shell script run.sh which runs the python script double-re-short.py (see README)
```
ERROR: Cannot open file restart_file: No such file or directory (src/read_data.cpp:367)
Last command: read_data       "restart_file"
```
PACKAGES/grem/lj-6rep/in.gREM-tempter: Need to run the shell script run.sh which runs the python script double-re-short.py (see README)
```
ERROR: Cannot open file restart_file: No such file or directory (src/read_data.cpp:367)
Last command: read_data       "restart_file"
```
PACKAGES/hnnp: needs -DPKG_ML-HDNNP=on (skipped)
PACKAGES/interlayer/kolmogorov_crespi_full/in.bilayer-graphene: completed but thermo output columns changed in the input file vs that in the reference log file
PACKAGES/interlayer/aip_water_2dm/in.gr_water:
```
ERROR on proc 0: cannot open aip/water/2dm potential file COH.aip.water.2dm: No such file or directory (src/potential_file_reader.cpp     :58)
Last command: pair_coeff  * *   aip/water/2dm        COH.aip.water.2dm C Ow Hw  # C-H2O
```
PACKAGES/interlayer/aip_water_2dm/in.gr_water.opt:
```
ERROR on proc 0: cannot open aip/water/2dm potential file COH.aip.water.2dm: No such file or directory (src/potential_file_reader.cpp     :58)
Last command: pair_coeff  * *   aip/water/2dm        COH.aip.water.2dm C Ow Hw  # C-H2O
```
PACKAGES/ipi: `ERROR on proc 0: Error opening UNIX socket: server may not be running or the path to the socket unavailable (src/MISC/fix_ipi.cpp:129)`
PACKAGES/latboltz, needs -DPKG_LATBOLTZ=on (skipped)
PACKAGES/latboltz/toycar/in.toycar: typo in the input script
`ERROR: Expected floating point parameter instead of '-7.5.0' in input script or data file (src/velocity.cpp:424)` Last command: velocity all set 0.0 -7.5.0 0.0 units box
PACKAGES/local_density/benzene_water/in.benzene_water: completed but no Step in thermo output (need to include Step)
PACKAGES/local_density/methanol_implicit_water/in.methanol_implicit_water: completed but no Step in thermo output (need to include Step)
PACKAGES/machdyn/fluid_structure_interaction/in.fluid_structure_interaction:
`ERROR: Must set 'extscalar' when setting 'scalar_flag' for compute smd/vol.  Contact the developer. (src/compute.cpp:122)`
PACKAGES/machdyn/aluminum_strip_pull/in.aluminum_strip_pull:
`ERROR on proc 0: Atoms have moved too far apart (4.740734222595944e+131) for minimum image`
PACKAGE/mgpt: needs -DPKG_MGPT=on (skipped)
PACKAGES/pace: needs -DPKG_ML-PACE=on (skipped)
PACKAGES/pafi: in.pafi, `ERROR on proc 0: cannot open eam/fs potential file ../../../../potentials/Fe_mm.eam.fs: No such file or directory (src/potential_file     _reader.cpp:58)` (need to create symlinks to those in ../../potentials, like in manybody examples)
PACKAGES/phonon/2-1D-diatomic: in.disp is not LAMMPS input scripts (need to rename, or skip in the config.yaml file)
PACKAGES/phonon/3-3D-FCC-Cu-EAM: in.disp2 is not LAMMPS input scripts (need to rename, or skip in the config.yaml file)
PACKAGES/phonon/dynamical_matrix_command/Silicon/in.silicon: completed, but no Step or Loop in the log file
PACKAGES/rann: needs -DPKG_ML-RANN=on (skipped)
PACKAGES/qtb: needs -DPKG_QTB=on (skipped)
PACKAGES/quip: needs -DPKG_ML-QUIP=on (skipped)
PACKAGE/mgpt: needs -DPKG_MGPT=on (skipped)
PACKAGES/scafacos: -DPKG_SCAFACOS=on (skipped)
PACKAGES/slater/in.slater takes long with mpirun -np 1, no reference log file -- need a separate run
PACKAGES/smqtb: needs -DPKG_SMQTB=on (skipped)
PACKAGES/ti/in.ti_spring:
```
ERROR on proc 0: cannot open eam/alloy potential file ../../../../potentials/Cu_mishin1.eam.alloy: No such file or directory (src/pot     ential_file_reader.cpp:58)
Last command:   pair_coeff        * * ../../../../potentials/Cu_mishin1.eam.alloy Cu
```
PACKAGES/vtk: needs -DPKG_VTK=on (skipped)

**Comment by akohlmey** [comment]
@ndtrung81 this is great. One issue is now fixed in commit 462e9756ce066b3a6a95175c6bf837e3d43c4ef8

**Comment by akohlmey** [comment]
@srtee may want to look at the ELECTRODE package issues, which looks a lot like it could be an off-by-one issue

**Comment by akohlmey** [comment]
> grid/in.grid.2d and in.sph

@ndtrung81 @sjplimp I am wondering if this could be related to the observation in issue #4264

**Comment by akohlmey** [comment]
> template/in.mol-restart-mix: no molecular-mix.restart, need to write out from the run with in.molecular-mix
> template/in.tmpl-restart-mix: template-mix.restart, need to write out from the run with in.template-mix

Made some modification to the inputs to first generate the restart and then delete it at the end. See commit 8a77949db22c5a96d81b9dd55ec620b6df77434d

**Comment by akohlmey** [comment]
> PACKAGES/ipi: `ERROR on proc 0: Error opening UNIX socket: server may not be running or the path to the socket unavailable (src/MISC/fix_ipi.cpp:129)`

FYI, fix ipi is like fix imd only that it connects to the i-pi software instead of VMD

**Comment by akohlmey** [comment]
@ndtrung81 been going through the list and fixed up what was easy to do (symlinks, create reference logs, small edits).

For the "extscalar" errors, you better talk to @sjplimp or - even better - report them as bug report issue and assign them to him. ;-)

**Comment by ndtrung81** [comment]
thanks @akohlmey for the quick fixes!

**Comment by akohlmey** [comment]
> mliap/in.mliap.ace.compute
> 
> ```
> ERROR: Illegal compute mliap command (src/ML-IAP/compute_mliap.cpp:110)
> Last command: compute     ace all mliap descriptor ace H_N_O_ccs.yace model linear gradgradflag 1
> ```

This requires installing the ML-PACE package and enabling ACE support in ML-IAP. In CMake that means, `-D PKG_ML-PACE=on -D PKG_ML-IAP=on -D MLIAP_ENABLE_ACE=on`.

I've updated the error messages to be more informative (some even said the error was with the pair style...).

**Comment by ndtrung81** [comment]
ASPHERE: in.line `Lost atoms` may be due to some changes in the parameters specified for the `pair coeff` command. The log file is 10 years old now. I can modify the in.line input script and add a new log file.
ASPHERE: in.line.srd `Lost atoms` the needed change seems more involving as the pair coeff params (sigma for 1-2 pairs) need to be right to push small particles out of the big ones, presumably the `delete_atoms` command works properly. Still trying to work it out.

**Comment by akohlmey** [comment]
@ndtrung81 just pushed fixes for LATBOLZ and MACHDYN to PR #4275

**Comment by akohlmey** [comment]
> PACKAGES/electrode/: in.conp, in.conq, in.conq2, in.etypes, in.ffield, in.ramp, in.thermo AddressSanitizer reported errors

fixed in a8447b419d5b78e56364896ddb2cf693951ca5b6 thanks to @srtee identifying the issue.

**Review by akohlmey — APPROVED** [review]

**Pull request merged**